client/services/FeedbackGeneratorService.js

```diff
@@ -53,94 +55,102 @@ tie.factory('FeedbackGeneratorService', [
+            var allowedOutputExample = (
+              correctnessTests[i].getAnyAllowedOutput());
+            var feedback = FeedbackObjectFactory.create(false);
+            feedback.appendTextParagraph([
+              'Your code gave the output ',
+              _jsToHumanReadable(observedOutput),
+              ' for the input ',
+              _jsToHumanReadable(correctnessTests[i].getInput()),
+              ' ... but this does not match the expected output ',
+              _jsToHumanReadable(allowedOutputExample),
+              '.'
+            ].join(''));
+            return feedback;
           }
+        }
 
-          var performanceTests = prompt.getPerformanceTests();
-          var performanceTestResults =
-              codeEvalResult.getPerformanceTestResults();
-          for (var i = 0; i < performanceTests.length; i++) {
-            var expectedPerformance = performanceTests[i].getExpectedPerformance();
-            var observedPerformance = performanceTestResults[i];
+        var performanceTests = prompt.getPerformanceTests();
+        var performanceTestResults =
+            codeEvalResult.getPerformanceTestResults();
+        for (var i = 0; i < performanceTests.length; i++) {
+          var expectedPerformance = performanceTests[i].getExpectedPerformance();
+          var observedPerformance = performanceTestResults[i];
 
-            if (expectedPerformance !== observedPerformance) {
-              var feedback = FeedbackObjectFactory.create(false);
-              feedback.appendTextParagraph([
-                'Your code is running more slowly than expected. Can you ',
-                'reconfigure it such that it runs in ',
-                expectedPerformance,
-                ' time?'
-              ].join(''));
-              return feedback;
-            }
+          if (expectedPerformance !== observedPerformance) {
+            var feedback = FeedbackObjectFactory.create(false);
+            feedback.appendTextParagraph([
+              'Your code is running more slowly than expected. Can you ',
+              'reconfigure it such that it runs in ',
+              expectedPerformance,
+              ' time?'
+            ].join(''));
+            return feedback;
           }
-
-          var feedback = FeedbackObjectFactory.create(true);
-          feedback.appendTextParagraph([
-            'You\'ve completed all the tasks for this question! Click the ',
-            '"Next" button to move on to the next question.',
-          ].join(''));
-          return feedback;
         }
+
+        var feedback = FeedbackObjectFactory.create(true);
+        feedback.appendTextParagraph([
+          'You\'ve completed all the tasks for this question! Click the ',
+          '"Next" button to move on to the next question.',
+        ].join(''));
+        return feedback;
+      }
+    };
+
+    return {
+      _prepareFeedback: _prepareFeedback,
+      getFeedback: function(prompt, codeEvalResult) {
```

Comment:
So I think it makes more sense to do the Transcript processing stuff here, but that's not possible for getSyntaxErrorFeedback since CodeEvalResult isn't passed in, so I have to handle it at the Solution level. Does it make more sense to try to pull CodeEvalResult into getSyntaxErrorFeedback, to push all the TranscriptService stuff up to the SolutionLevel, or to handle each individually? Happy to fix it to what's best.